### ISSUE TITLE
Add "servo" wrapper class to SoftPWMServo

### ISSUE THREAD

**Issue by EmbeddedMan** [chipKIT32/chipKIT32-MAX#575]
The idea here is that it will have exactly the same interface as the standard Arduino servo class, so you can just drop in SoftPWMServo.h instead of Servo.h and all existing sketches will just work.

**Comment by majenkotech** [comment]
You could inherit the Servo class as a parent then and override the relevant functions.  That way the SoftPWMServo class can be passed to any functions that expect a Servo object.  Means you have to include both Servo and SoftPWMServo (if you're not on UECIDE of course)...

**Comment by EmbeddedMan** [comment]
True. I've got everything done, for now, in a simple way. Next step is to
make it cleaner, as you suggest.

**Comment by majenkotech** [comment]
Personally I'd like to see more interface classes defined in the core - things like the Servo interface, etc, than the libraries then use instead of defining their own interfaces.  Makes for a much cleaner system.

**Comment by EmbeddedMan** [comment]
I completely agree. The trick, of course, is doing it without affecting
existing sketches.

**Comment by majenkotech** [comment]
Yeah, Arduino is a real millstone round our necks...

**Comment by ricklon** [comment]
Confirmed closed?

**Comment by EmbeddedMan** [comment]
This should be confirmed by somebody other than me before being closed. At least test it out that it works for them.
